src/data/part.lean

```diff
@@ -185,11 +191,14 @@ by cases o; refl
 @[simp] theorem of_to_option (o : part α) [decidable o.dom] : of_option (to_option o) = o :=
 ext $ λ a, mem_of_option.trans mem_to_option
 
+/-- When the domains of all elements in `part α` are decidable, `part α` is equivalent to
+`option α`. Proven unconditionally by going classical. -/
```

Comment:
Actually, this can't be proven at all without going classical, because we're quantifying over all elements of `part A`. I'm pretty sure this theorem is equivalent to LEM on any nonempty type.

Reply:
I have this intuition too, but I'm not so sure... Let's change it anyway.